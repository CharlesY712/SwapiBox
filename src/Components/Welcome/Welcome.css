.welcome {
  width: 1000px;
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  top: 0px;
  height: 430px;
  width: 105%;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 300%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 600px;
  text-align: justify;
  overflow: hidden;
}

.crawl {
  margin-top: 450px;
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 90s linear;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(30deg)  translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(15deg) translateZ(-2500px);
  }
}