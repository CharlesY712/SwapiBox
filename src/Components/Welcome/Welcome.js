import React from "react";
import PropTypes from 'prop-types';
import "./Welcome.css";


const Welcome = ({crawl}) => {
  return (
    <div className="welcome">
      <div className="fade"></div>
      <section className="star-wars">
        <div className="crawl">
          <div className="title">
            <p>Episode {crawl.episode}</p>
            <h1>{crawl.title}</h1>
          </div>
          <p>{crawl.crawl}</p>
        </div>
      </section>
    </div>
  );
};

export default Welcome;

Welcome.propTypes = {
  crawl: PropTypes.object
};