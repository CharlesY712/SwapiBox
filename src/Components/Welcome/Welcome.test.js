import React from 'react';
import Welcome from "./Welcome";
import {shallow} from "enzyme";
import mockOpeningCrawlData from "../../MockData/MockOpeningCrawl";

describe('Welcome', () => {
  const mockCrawlData = mockOpeningCrawlData;
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Welcome crawl={mockCrawlData}/>);
  });
  
  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});