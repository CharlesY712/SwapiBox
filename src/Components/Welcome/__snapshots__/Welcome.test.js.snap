// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    crawl={
        Object {
            "crawl": "It is a period of civil war.
        Rebel spaceships, striking
        from a hidden base, have won
        their first victory against
        the evil Galactic Empire.
        
        During the battle, Rebel
        spies managed to steal secret
        plans to the Empire's
        ultimate weapon, the DEATH
        STAR, an armored space
        station with enough power
        to destroy an entire planet.
        
        Pursued by the Empire's
        sinister agents, Princess
        Leia races home aboard her
        starship, custodian of the
        stolen plans that can save her
        people and restore
        freedom to the galaxy....",
            "episode": 4,
            "release": "1977",
            "title": "A New Hope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="star-wars"
>
        <div
                className="crawl"
        >
                <div
                        className="title"
                >
                        <p>
                                Episode 
                                4
                                  (
                                1977
                                )
                        </p>
                        <h1>
                                A New Hope
                        </h1>
                </div>
                <p>
                        It is a period of civil war.
                        Rebel spaceships, striking
                        from a hidden base, have won
                        their first victory against
                        the evil Galactic Empire.
                        
                        During the battle, Rebel
                        spies managed to steal secret
                        plans to the Empire's
                        ultimate weapon, the DEATH
                        STAR, an armored space
                        station with enough power
                        to destroy an entire planet.
                        
                        Pursued by the Empire's
                        sinister agents, Princess
                        Leia races home aboard her
                        starship, custodian of the
                        stolen plans that can save her
                        people and restore
                        freedom to the galaxy....
                </p>
        </div>
</section>,
      "className": "welcome",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
>
          <div
                    className="title"
          >
                    <p>
                              Episode 
                              4
                                (
                              1977
                              )
                    </p>
                    <h1>
                              A New Hope
                    </h1>
          </div>
          <p>
                    It is a period of civil war.
                    Rebel spaceships, striking
                    from a hidden base, have won
                    their first victory against
                    the evil Galactic Empire.
                    
                    During the battle, Rebel
                    spies managed to steal secret
                    plans to the Empire's
                    ultimate weapon, the DEATH
                    STAR, an armored space
                    station with enough power
                    to destroy an entire planet.
                    
                    Pursued by the Empire's
                    sinister agents, Princess
                    Leia races home aboard her
                    starship, custodian of the
                    stolen plans that can save her
                    people and restore
                    freedom to the galaxy....
          </p>
</div>,
        "className": "star-wars",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              <p>
                            Episode 
                            4
                              (
                            1977
                            )
              </p>
              <h1>
                            A New Hope
              </h1>
</div>,
            <p>
              It is a period of civil war.
              Rebel spaceships, striking
              from a hidden base, have won
              their first victory against
              the evil Galactic Empire.
              
              During the battle, Rebel
              spies managed to steal secret
              plans to the Empire's
              ultimate weapon, the DEATH
              STAR, an armored space
              station with enough power
              to destroy an entire planet.
              
              Pursued by the Empire's
              sinister agents, Princess
              Leia races home aboard her
              starship, custodian of the
              stolen plans that can save her
              people and restore
              freedom to the galaxy....
</p>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Episode 
                  4
                    (
                  1977
                  )
</p>,
                <h1>
                  A New Hope
</h1>,
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Episode ",
                    4,
                    "  (",
                    "1977",
                    ")",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Episode ",
                  4,
                  "  (",
                  "1977",
                  ")",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A New Hope",
                },
                "ref": null,
                "rendered": "A New Hope",
                "type": "h1",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
            },
            "ref": null,
            "rendered": "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <p>
                                        Episode 
                                        4
                                          (
                                        1977
                                        )
                              </p>
                              <h1>
                                        A New Hope
                              </h1>
                    </div>
                    <p>
                              It is a period of civil war.
                              Rebel spaceships, striking
                              from a hidden base, have won
                              their first victory against
                              the evil Galactic Empire.
                              
                              During the battle, Rebel
                              spies managed to steal secret
                              plans to the Empire's
                              ultimate weapon, the DEATH
                              STAR, an armored space
                              station with enough power
                              to destroy an entire planet.
                              
                              Pursued by the Empire's
                              sinister agents, Princess
                              Leia races home aboard her
                              starship, custodian of the
                              stolen plans that can save her
                              people and restore
                              freedom to the galaxy....
                    </p>
          </div>
</section>,
        "className": "welcome",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <p>
                                    Episode 
                                    4
                                      (
                                    1977
                                    )
                        </p>
                        <h1>
                                    A New Hope
                        </h1>
            </div>
            <p>
                        It is a period of civil war.
                        Rebel spaceships, striking
                        from a hidden base, have won
                        their first victory against
                        the evil Galactic Empire.
                        
                        During the battle, Rebel
                        spies managed to steal secret
                        plans to the Empire's
                        ultimate weapon, the DEATH
                        STAR, an armored space
                        station with enough power
                        to destroy an entire planet.
                        
                        Pursued by the Empire's
                        sinister agents, Princess
                        Leia races home aboard her
                        starship, custodian of the
                        stolen plans that can save her
                        people and restore
                        freedom to the galaxy....
            </p>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <p>
                                Episode 
                                4
                                  (
                                1977
                                )
                </p>
                <h1>
                                A New Hope
                </h1>
</div>,
              <p>
                It is a period of civil war.
                Rebel spaceships, striking
                from a hidden base, have won
                their first victory against
                the evil Galactic Empire.
                
                During the battle, Rebel
                spies managed to steal secret
                plans to the Empire's
                ultimate weapon, the DEATH
                STAR, an armored space
                station with enough power
                to destroy an entire planet.
                
                Pursued by the Empire's
                sinister agents, Princess
                Leia races home aboard her
                starship, custodian of the
                stolen plans that can save her
                people and restore
                freedom to the galaxy....
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode 
                    4
                      (
                    1977
                    )
</p>,
                  <h1>
                    A New Hope
</h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      4,
                      "  (",
                      "1977",
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    4,
                    "  (",
                    "1977",
                    ")",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
              },
              "ref": null,
              "rendered": "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
